src/ai/PathFinder.cpp

```diff
@@ -151,29 +151,25 @@ class PathFinder::OpenNodeList {
 
 class PathFinder::ClosedNodeList {
 	
-	typedef std::vector<Node*> NodeList;
-	NodeList nodes;
+	typedef std::vector<bool> NodeList;
+	NodeList *nodes;
 	
 public:
+
+	ClosedNodeList(int nodeCount) {
+		nodes = new NodeList(nodeCount, false);
+	}
 	
 	~ClosedNodeList() {
-		for(NodeList::iterator i = nodes.begin(); i != nodes.end(); ++i) {
-			delete *i;
-		}
+		delete nodes;
 	}
 	
 	void add(Node * node) {
```

Comment:
If the implementation stays like this, this can be changed into a `const Node *`.